### ISSUE TITLE
Incorrect construction of SimpleArray from a discontiguous slice of ndarray

### ISSUE THREAD

**Issue by xingularity** [solvcon/solvcon#432]
When a SimpleArray is constructed through the "array" argument in Python and the numpy array passed to this argument is a subset using ndarray accessor, the SimpleArray is constructed with wrong section of data from the original ndarray. Following code is a demonstration.

```
from modmesh import SimpleArrayFloat64
import numpy as np
a = np.array([[1,2,3,4,5], [10,20,30,40,50],[100,200,300,400,500]])
```
Now we have 

```
array([[  1,   2,   3,   4,   5],
       [ 10,  20,  30,  40,  50],
       [100, 200, 300, 400, 500]])
```
If we construct a SimpleArray with
```
b = SimpleArrayFloat64(array=a[:, 4])
print(b.ndarray)
```
The print result is [ 5., 10., 20.].
We should expect a SimpleArray contains [5, 50, 500] instead of [ 5., 10., 20.]

**Comment by yungyuc** [comment]
The `SimpleArray` constructor expects a contiguous memory buffer and disregard the strides set in `ndarray`.  Before `SimpleArray` supports the same stride as `ndarray`, it should throw an exception when the input `ndarray` is not contiguous.

For now, please use `b = SimpleArrayFloat64(array=a[:,4].copy())` to work around.

**Comment by xingularity** [comment]
Thanks @yungyuc, I already used that workaround in my code.

**Comment by yungyuc** [comment]
@tigercosmos Would you have time to take a look?

**Comment by tigercosmos** [comment]
I can help support this starting from the beginning of November.

**Comment by yungyuc** [comment]
Fixed by #438
